### ISSUE TITLE
FFI/LibCurl buffer overflow under heavy load.

### ISSUE THREAD

**Issue by jgwmaxwell** [jruby/jruby#752]
I mentioned this as part of #231 but reopening as a new issue.
#### Situation:

Server: 

```
TorqueBox 3.x.incremental.1606
```

JRuby: 

```
1.7.3
```

Java (Oracle):

```
java version "1.7.0_21"
Java(TM) SE Runtime Environment (build 1.7.0_21-b11)
Java HotSpot(TM) 64-Bit Server VM (build 23.21-b01, mixed mode)
```

Affected Gems:

```
Typhoeus 0.6.3
Ethon 0.5.12
Faraday 0.8.7
```
#### Stack Trace:

[https://gist.github.com/jgwmaxwell/5621074](https://gist.github.com/jgwmaxwell/5621074)
#### Outline:

Running Typhoeus using the #in_parallel request grouping method in a TorqueBox 'Processor' works fine, until the load gets pretty high, then it does this. The problems come when I've got 10 threads, dispatching 3-5 requests in parallel each, running approximately 1.5k 'jobs' per minute between them, or 4.5k-7.5k req/min.

The server is fine on all RAM meters, isn't swapping, and CPU is sitting around 85% (this is also serving web requests, servicing STOMP clients and doing a few other things at the same time).

It's not a production critical issue for us, as we can simply swap Typhoeus out, but thought it might be interesting to raise!

**Comment by enebo** [comment]
Due to age and the fact I ran repo from last reporter without seeing a crash I will assume this is working or just so old it is not triggering (although I am using JRuby 9k instead of 1.7.x since we are EOL'ing 1.7.x soon).  If someone has more knowledge on this issue then please open a new issue on it.
